common/thorhelper/thorsoapcall.cpp

```diff
@@ -2231,9 +2231,9 @@ class CWSCAsyncFor : implements IWSCAsyncFor, public CInterface, public CAsyncFo
             {
                 try
                 {
-                    checkTimeLimitExceeded(&remainingMS);
                     Url &connUrl = master->proxyUrlArray.empty() ? url : master->proxyUrlArray.item(0);
                     ep.set(connUrl.host.get(), connUrl.port);
+                    checkTimeLimitExceeded(&remainingMS);  // after ep.set which might make a potentially long getaddrinfo lookup ...
```

Comment:
ok, I can add a checkTimeLimitExceeded() at its orig location so its checked there and also after the getaddrinfo() from the ep.set().
The connect() using master->timeoutMS change to use remainingMS is debatable because previously it would have done a single connect with the timeout set to number of attempts * timeout - so to be the same we use the timeout for each connect.
Similarly, there is some discussion if we should checkTimeLimitExceeded() in the connect() retry loop so that multiple connect() attempts do not exceed a timeLimit if provided.  Or should timeOut and timeLimit be completely separate ?
So I think this is a good suggestion and perhaps worth of its own JIRA to discuss, but think it should remain as it is here.

Reply:
I agree with both, keep original checkTimeLimitExceeded(&remainingMS); as well as this new line.

And open a new JIRA to investigate applying time limit to connect (in addition to timeout).

Reply:
I've added back a checkTimeLimitExceeded() at its original location in commit #2.
@jakesmith @afishbeck if ok with this I will squash.

Reply:
agreed, can you link the new JIRA to this JIRA? - to make it a little easier to find. Thanks.